canine/localization/base.py

```diff
@@ -364,28 +364,50 @@ def delocalize(self, patterns: typing.Dict[str, str], output_dir: str = 'canine_
                     output_files[jobId][outputname] = [dirpath]
         return output_files
 
-    def pick_common_inputs(self, inputs: typing.Dict[str, typing.Dict[str, str]], overrides: typing.Dict[str, typing.Optional[str]], transport: typing.Optional[AbstractTransport] = None) -> typing.Dict[str, str]:
+    def flatten_inputs(self, obj: typing.Any) -> typing.Generator[str, None, None]:
+        """
+        Recurses over any object given to yield the flattened elements.
+        Used to pick out the filepaths which will be commonized
+        """
+        if isinstance(obj, dict):
+            for val in obj.values():
+                yield from self.flatten_inputs(val)
+        elif isinstance(obj, list):
+            for val in obj:
+                yield from self.flatten_inputs(val)
+        else:
+            yield obj
+
+    def pick_common_inputs(self, inputs: typing.Dict[str, typing.Dict[str, typing.Union[str, typing.List[str]]]], overrides: typing.Dict[str, typing.Optional[str]], transport: typing.Optional[AbstractTransport] = None) -> typing.Dict[str, str]:
         """
         Scans input configuration and overrides to choose inputs which should be treated as common.
         Returns the dictionary of common inputs {input path: common path}
         """
         with self.transport_context(transport) as transport:
-            self.common_inputs = set()
             seen = set()
+            self.common_inputs = set()
+            seen_forced = set()
             for jobId, values in inputs.items():
-                for arg, path in values.items():
-                    if path in seen and (arg not in overrides or overrides[arg] == 'common'):
-                        self.common_inputs.add(path)
-                    if arg in overrides and overrides[arg] == 'common':
-                        self.common_inputs.add(path)
-                    seen.add(path)
+                for arg, paths in values.items():
+                    if arg not in overrides:
+                        print(arg, "is not overridden; scanning inputs")
```

Comment:
(When we implement variable-verbosity outputs, stuff like this will be super low tier.)